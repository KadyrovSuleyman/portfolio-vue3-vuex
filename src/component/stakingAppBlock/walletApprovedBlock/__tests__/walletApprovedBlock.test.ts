import { mount, VueWrapper } from '@vue/test-utils';
import WalletApprovedBlock from '../walletApprovedBlock.vue';

let wrapper: VueWrapper<any>;
afterEach(() => {
  wrapper.unmount();
});

it('walletApprovedBlock renders', () => {
  wrapper = mount(WalletApprovedBlock);
  expect(wrapper.element.outerHTML).toMatchSnapshot();
});

it('watchs props changes', async () => {
  wrapper = mount(WalletApprovedBlock);
  expect(wrapper.find('.walletApprovedBlock').classes()).toEqual(['walletApprovedBlock']);

  await wrapper.setProps({
    mods: {
      selected: true,
    },
  });
  expect(wrapper.find('.walletApprovedBlock').classes()).toEqual(['walletApprovedBlock', 'walletApprovedBlock__selected']);

  await wrapper.setProps({
    mods: {
      selected: false,
    },
  });
  expect(wrapper.find('.walletApprovedBlock').classes()).toEqual(['walletApprovedBlock']);
});
