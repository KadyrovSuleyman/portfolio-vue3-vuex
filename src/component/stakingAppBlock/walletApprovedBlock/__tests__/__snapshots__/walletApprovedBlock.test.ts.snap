// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`walletApprovedBlock renders 1`] = `
<div class="walletApprovedBlock"><span class="walletApprovedBlock-span"> Wallet approved </span>
  <div class="walletApprovedBlock-div walletApprovedBlock-div__icon_check"></div>
</div>
`;
